Color + button refactoring and some feedback

Charlesthebird opened the pull request:
This PR:
- Updates the colors file to use the colors NPM package more consistently and use variables for base colors.
- Adds the filter by rejected option to the admin subscriptions page.
- Removes the no results animation so that it's just text based on feedback that the animation may indicate something is loading.
- Updates the `Common/Button` component to use a themed Mantine button.

<img width="1322" alt="Screenshot 2024-04-09 at 2 05 29 PM" src="https://github.com/solo-io/dev-portal-starter/assets/4720646/4e284c34-9e11-461c-8f1c-a91c812bb832">

Charlesthebird commented:
Issue: https://github.com/solo-io/gloo-mesh-enterprise/issues/14998

Worfle reviewed (approved):
These feel like a direction rather than a need.  I'm not challenging, but wondering where we are in the process of updating this repo.

Thoughts?

The work here looks ready to go in though

Charlesthebird commented:
@Worfle That's true, I'll add some context here.

Functionally the repo is ready for 2.6 when the backend is. The backend work is paused for the foreseeable future - this is the last week of UI work on it so I might have 1-2 more PR's wrapping up the rest of the refactoring ideas I noted [here](https://github.com/solo-io/gloo-mesh-enterprise/issues/14998).

A lot of these changes are for consistency (with the changes I introduced starting in the Sass -> emotion refactor), to make it easier to pick back up when we revisit it eventually.

The Mantine button had some usefulness regarding the variants + colors + hover/active states and how it was utilizing the library theme. I added some of these buttons earlier, and wanted to make it consistent across the board.

The rest of the changes were from the Portal meeting today where I demoed this, which was the last meeting until the backend work resumes.
